### PR TITLE
Fix https://github.com/Seneca-CDOT/telescope/issues/1930

### PR DESCRIPTION
**Pull request by humphd**
Fixes https://github.com/humphd/satellite/pull/new/telescope-issue-1930

This adds a new function to the `isAuthorized()` middleware, allowing arbitrary comparisons with a user's claims in their bearer token.  This can be used to compare the user's id (i.e., `sub`) or examine `roles`, etc.

I think that between having `roles` and `authorizeUser()`, it's pretty easy to create any scenario we might want.

### CODE REVIEW

File: src/middleware.js [hunk 1]
```diff
@@ -14,29 +14,72 @@ function isAuthenticated() {
...
       }
+
+      // If defined, check that the user's payload data matches what's expected
+      if (authorizeUser) {
+        if (!authorizeUser(user)) {
+          next(createError(403, `user not authorized`));
+          return;
+        }
+      }
+    } catch (err) {
```
Comment (Metropass) [inline src/middleware.js:79]: Wouldn't `err` be of type `http-error`, if so, couldn't we use `err.message` instead of `"unexpected error"`?

Comment (humphd) [reply src/middleware.js:79]: We'll log that with the `{ err }` bit.  The 'unexpected error' part refers that this error shouldn't have happened (it's not something we should ever see, so I want to flag that).

Comment (Metropass) [reply src/middleware.js:79]: Ah, I see, that makes sense.

**Review by Metropass — APPROVED** [review]
They work locally on my machine, with all tests passed

**Review by HyperTHD — APPROVED** [review]

**Review by chrispinkney — APPROVED** [review]

**Pull request merged**